workers/unity/Packages/com.improbable.gdk.transformsynchronization/TransformUtils.cs

```diff
@@ -11,51 +11,146 @@ public static bool HasChanged(Coordinates a, Coordinates b)
         }
 
         // Checking for no change, so exact equality is fine
-        public static bool HasChanged(Location a, Location b)
+        public static bool HasChanged(CompressedQuaternion a, CompressedQuaternion b)
         {
-            return a.X != b.X || a.Y != b.Y || a.Z != b.Z;
+            return a.Data != b.Data;
         }
 
         // Checking for no change, so exact equality is fine
-        public static bool HasChanged(Quaternion a, Quaternion b)
+        public static bool HasChanged(FixedPointVector3 a, FixedPointVector3 b)
+        {
+            return a.X != b.X || a.Y != b.Y || a.Z != b.Z;
+        }
+
+        // Decompress from uint32 to Unity Quaternion
+        public static UnityEngine.Quaternion ToUnityQuaternion(this CompressedQuaternion quaternion)
```

Comment:
Please add comments for these compressions. Someone coming to this without context will have to slog through it 😉